graph/cycle_detection.cpp

```diff
@@ -0,0 +1,126 @@
+#include <vector> //for std::vector<T>
+#include <iostream> //for IO operations
+
+// The algorithm is using the colouring technique for finding a cycle in a graph
+// you can see more details here : https://en.wikipedia.org/wiki/Graph_coloring
+
+
+class Graph {
+ private:
+  int V;//number of Vertices
+  std::vector<int> *adj;//adjacency list for our graph
+
+  //typical node colouring , if the node is unvisited we color it as NOT_STARTED
+  //if we've started the traversal , we color it as STARTED
+  //if we've finished the traversal we color it as FINISHED
```

Comment:
//if the node is unvisited, we color it as NOT_STARTED
//if we've finished the traversal, we color it as FINISHED